src/services/stringCompletions.ts

```diff
@@ -130,7 +130,15 @@ namespace ts.Completions.StringCompletions {
                         //          bar: string;
                         //      }
                         //      let x: Foo["/*completion position*/"]
-                        return stringLiteralCompletionsFromProperties(typeChecker.getTypeFromTypeNode((grandParent as IndexedAccessTypeNode).objectType));
+                        const { objectType } = grandParent as IndexedAccessTypeNode;
+                        let isNodeAnObjectTypeStringLiteral;
+                        if((objectType as any).literal){
+                            isNodeAnObjectTypeStringLiteral = node.text === (objectType as any).literal.text;

```

Comment:
Don't check the text, check if `parent !== grandParent.indexType` and just return `undefined`.

Reply:
@DanielRosenwasser I applied your suggestion and I also added a check for parenthesized types.